IPython/core/display.py

```diff
@@ -1059,6 +1078,9 @@ def __init__(self, data=None, url=None, filename=None, format=None,
                     format = self._FMT_JPEG
                 if ext == u'png':
                     format = self._FMT_PNG
+                if ext == u'gif':
+                    # use PNG format until we understand why GIF is not working
+                    format = self._FMT_PNG
```

Comment:
It's in ipykernel, which is a rather icky bit of tight coupling. :-(

https://github.com/ipython/ipykernel/blob/8acaee80c83c9a64a22afde522f9df2d0c3638e4/ipykernel/jsonutil.py#L51